core/test/lib/workarea/ext/freedom_patches/dragonfly_job_fetch_url_test.rb

```diff
@@ -0,0 +1,24 @@
+require 'test_helper'
+
+module Workarea
+  class DragonflyJobFetchURLTest < TestCase
+    def test_fail_early_on_misconfigured_proxy
+      product = create_product
+
+      stub_request(:any, 'http://via.placeholder.com/350x150')
```

Comment:
This is just [Webmock](https://github.com/bblimke/webmock), used directly, in order to stub out the request to `via.placeholder.com`. I couldn't find any 3rd-party services that would return me a custom status code _and_ headers, but I kinda like this solution better anyway as it doesn't rely on anything we weren't depending on before.